contracts/SBCDepositContract.sol

```diff
@@ -224,4 +242,211 @@ contract SBCDepositContract is IDepositContract, IERC165, IERC677Receiver, Pausa
+        emit FailedWithdrawalProcessed(_failedWithdrawalId, amountToProceed, failedWithdrawalRecord.receiver);
+    }
+
+    uint256 public failedWithdrawalsPointer;
+
+    /**
+     * @dev Function to be used to process failed withdrawals.
+     * Call to this function will revert only if it ran out of gas or it is a reentrant access to failed withdrawals processing.
+     * Call to this function doesn't transmit flow control to any untrusted contract and uses a constant gas limit for each withdrawal,
+     * so using constant gas limit and constant max number of withdrawals for calls of this function is ok.
+     * @param _maxNumberOfFailedWithdrawalsToProcess Maximum number of failed withdrawals to be processed.
+     */
+    function processFailedWithdrawalsFromPointer(uint256 _maxNumberOfFailedWithdrawalsToProcess)
+        public
+        failedWithdrawalProcessNonReentrant
+    {
+        for (uint256 i = 0; i < _maxNumberOfFailedWithdrawalsToProcess; ++i) {
+            if (failedWithdrawalsPointer == numberOfFailedWithdrawals) {
+                break;
+            }
+
+            FailedWithdrawalRecord storage failedWithdrawalRecord = failedWithdrawals[failedWithdrawalsPointer];
+            if (!failedWithdrawalRecord.processed) {
+                bool success = _processWithdrawal(
+                    failedWithdrawalRecord.amount,
+                    failedWithdrawalRecord.receiver,
+                    onWithdrawalsUnwrapToGNOByDefault,
+                    DEFAULT_GAS_PER_WITHDRAWAL
+                );
+                if (!success) {
+                    break;
+                }
+                failedWithdrawalRecord.processed = true;
+                emit FailedWithdrawalProcessed(
+                    failedWithdrawalsPointer,
+                    failedWithdrawalRecord.amount,
+                    failedWithdrawalRecord.receiver
+                );
+            }
+
+            ++failedWithdrawalsPointer;
+        }
+    }
+
+    /**
+     * @dev Function to be used only in the system transaction.
+     * Call to this function will revert only in three cases:
+     *     - the caller is not `SYSTEM_WITHDRAWAL_EXECUTOR` or `_admin()`;
+     *     - the length of `_amounts` array is not equal to the length of `_addresses` array;
+     *     - it is a reentrant access to failed withdrawals processing;
+     *     - the call ran out of gas.
+     * Call to this function doesn't transmit flow control to any untrusted contract and uses a constant gas limit for each withdrawal,
+     * so using constant gas limit and constant number of withdrawals (including failed withdrawals) for calls of this function is ok.
+     * @param _maxNumberOfFailedWithdrawalsToProcess Maximum number of failed withdrawals to be processed.
+     * @param _amounts Array of amounts to be withdrawn.
+     * @param _addresses Array of addresses that should receive the corresponding amount of tokens.
+     */
+    function executeSystemWithdrawals(
```

Comment:
An idea to consider. If we would change this function to return a boolean flag indicating whether withdrawals were succeeded or not, an execution client could log a warning hinting to check the contract for details. This could improve the user experience a little at the cost of a slight gas usage increase because of the flag handling. Not entirely sure though whether it's worth that.

Reply:
Not sure of its value either, won't be opposed to it. It's not actionable info for a user, same way failed txs from the network are not logged.